src/libcollections/btree/node.rs

```diff
@@ -1397,6 +1474,41 @@ impl<BorrowType, K, V, HandleType>
     }
 }
 
+impl<'a, K, V> Handle<NodeRef<marker::Mut<'a>, K, V, marker::LeafOrInternal>, marker::Edge> {
+    /// Splits the node by the given edge (`self`) and puts the right part to `right_node`.
+    /// The first edge of `right_node` remains unchanged.
+    pub fn cut_to_right(&mut self,
```

Comment:
Nit: Another way to describe this operation would be moving a suffix from one node to another node. I'm not sure if `move_suffix` would be a clearer name, however.